From ae2cb8803c72456ebe4182084377e3ab15fc1e34 Mon Sep 17 00:00:00 2001
From: Jonathan David <jonathan.david@ni.com>
Date: Thu, 10 Jul 2014 12:31:39 -0500
Subject: [PATCH 5/7] Unpress key when mouse leaves key area

When user presses a key and moves the mouse to another key, the
key is lifted (preventing the key from being repeated in the active
input field). Behavior of keyboard is closer to that of a physical
keyboard.
---
 src/status.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/status.c b/src/status.c
index b503fbd..33d791b 100644
--- a/src/status.c
+++ b/src/status.c
@@ -261,6 +261,8 @@ void status_focus_set(struct status *status, struct key *focus)
 	view_update(status->view, status->focus, FALSE);
 	status->focus=focus;
 	view_update(status->view, status->focus, FALSE);
+	if(!status->moving)
+		status_pressed_set(status, NULL);
 }
 
 /* return the focus key */
-- 
2.0.1

